### PR TITLE
Updated the rest api docs for `deposits/overview-all`

### LINKED ISSUE #8346: Update `deposits/overview-all` API with `default_external_accounts` element.
### Description
The deposits `overview-all` API has been modified in https://github.com/Automattic/woocommerce-payments-server/pull/4862
The [REST API docs](https://github.com/Automattic/woocommerce-payments/blob/ca8c7c0d04400b6e4753e6baa919209e88752154/docs/rest-api/source/includes/wp-api-v3/deposits.md#L37) should be updated after the change is deployed

### Acceptance criteria
REST API docs updated.

### Additional context
https://github.com/Automattic/woocommerce-payments-server/pull/4862

### PR DESCRIPTION
**Pull request by jessy-p**
Fixes #8346

#### Changes proposed in this Pull Request
- Added `default_external_accounts` element to REST API docs for `deposits/overview-all`

#### Testing instructions
- Verify the markdown
- Can build docs locally to verify the REST API docs

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8364` or branch name `add/8346-deposits-overview-docs` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=add/8346-deposits-overview-docs) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 519c448df816c3c3b742cdba51e24978e00e2ef6
- Build time: 2024-03-14 06:30:02 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by naman03malhotra — APPROVED** [review]
Approving, looks good to me!

File: docs/rest-api/source/includes/wp-api-v3/deposits.md [hunk 1]
```diff
@@ -76,6 +76,9 @@ Fetch an overview of account deposits for all deposit currencies. This includes
         -   `weekly_anchor` _string_ | _undefined_ - The day of the week that payments are paid out, e.g. `monday`.
         -   `monthly_anchor` _int_ | _undefined_ - The day of the month that payments are paid out. Specified as a number between 1–31. 29-31 will instead use the last day of a shorter month.
     -   `default_currency` _string_ - The default currency for the account.
+	-   `default_external_accounts` _array_ - The default external accounts for the account.
```
Comment (Jinksi) [inline docs/rest-api/source/includes/wp-api-v3/deposits.md]: 💭 can we offer more context here?

E.g. mention what kind of account this is (deposit destination, not Stripe account) and what `default` means (max one per currency).

Comment (jessy-p) [reply docs/rest-api/source/includes/wp-api-v3/deposits.md]: How about 
_The default external payout acccounts (deposit destinations) for the Stripe account._

**Review by Jinksi — APPROVED** [review]
Looks good @jessy-p 💯 

I've checked this by running `build.sh` from `/docs/rest-api/` and viewing the changes to the docs ✅

File: docs/rest-api/source/includes/wp-api-v3/deposits.md (hunk 1 above)
Comment (Jinksi) [reply docs/rest-api/source/includes/wp-api-v3/deposits.md]: 💯 yeh nice. I added a small typo commit (acc~c~ounts) in 8b4cf9e.

Comment (Jinksi) [reply docs/rest-api/source/includes/wp-api-v3/deposits.md]: I hope you don't mind, I also wrapped the example values in backticks e.g. `errored` in 519c448 to keep it consistent with other example values, e.g. `monday`.

**Pull request merged**